src/manifolds/PositiveNumbers.jl

```diff
@@ -0,0 +1,224 @@
+            M::PositiveNumbers,
+            rm::AbstractRetractionMethod,
+        )
+    end,
+)
+
+@doc raw"""
+    inner(M::PositiveNumbers, p, X, Y)
+
+Compute the inner product of the two tangent vectors `X,Y` from the tangent plane at `p` on
+the [`PositiveNumbers`](@ref) `M`, i.e.
+
+````math
+g_p(X,Y) = \frac{XY}{p^2}.
+````
+"""
+inner(::PositiveNumbers, ::Any...)
+@inline inner(::PositiveNumbers, p::Real, X::Real, Y::Real) = X * Y / p^2
+
+function inverse_retract(M::PositiveNumbers, x::Number, y::Number)
+    return inverse_retract(M, x, y, LogarithmicInverseRetraction())
+end
+function inverse_retract(
+    M::PositiveNumbers,
+    x::Number,
+    y::Number,
+    ::LogarithmicInverseRetraction,
+)
+    return log(M, x, y)
+end
+
+@doc raw"""
+    log(M::PositiveNumbers, p, q)
+
+Compute the logarithmic map on the [`PositiveNumbers`](@ref) `M`.
+````math
+\log_p q = p\log\frac{q}{p}.
+````
+"""
+Base.log(::PositiveNumbers, p::Real, q::Real) = p * log(q / p)
+
+@doc raw"""
+    manifold_dimension(M::PositiveNumbers)
+
+Return the dimension of the [`PositiveNumbers`](@ref) `M`,
+i.e. of the 1-dimensional hyperbolic space,
+
+````math
+\dim(H^1) = 1
+````
+"""
+manifold_dimension(::PositiveNumbers) = 1
+
+mid_point(M::PositiveNumbers, p1, p2) = exp(M, p1, 0.5 * log(M, p1, p2))
+
+@inline LinearAlgebra.norm(::PositiveNumbers, p, X) = abs(X / p)
+
+@doc raw"""
+    project(M::PositiveNumbers, p, X)
+
+Project a value `X` onto the tangent space of the point `p` on the [`PositiveNumbers`](@ref) `M`,
+which is just the identity.
+"""
+project(::PositiveNumbers, ::Real, X::Real) = X
+
+retract(M::PositiveNumbers, p, q) = retract(M, p, q, ExponentialRetraction())
+retract(M::PositiveNumbers, p, q, ::ExponentialRetraction) = exp(M, p, q)
+
+representation_size(::PositiveNumbers) = ()
+
+sharp(::PositiveNumbers, ::Number, ξ::CoTFVector) = FVector(TangentSpace, ξ.data)
+
+Base.show(io::IO, ::PositiveNumbers) = print(io, "PositiveNumbers()")
+
+function Base.show(
+    io::IO,
+    ::PowerManifold{ℝ,PositiveNumbers,TSize,ArrayPowerRepresentation},
+) where {TSize}
+    s = [TSize.parameters...]
+    (length(s) == 1) && return print(io, "PositiveVectors($(s[1]))")
+    (length(s) == 2) && return print(io, "PositiveMatrices($(s[1]), $(s[2]))")
+    return print(io, "PositiveArrays($(join(s, ", ")))")
+end
+
+@doc raw"""
+    vector_transport_to(M::PositiveNumbers, p, X, q, ::ParallelTransport)
+
+Compute the parallel transport of `X` from the tangent space at `p` to the tangent space at
+`q` on the [`PositiveNumbers`](@ref) `M`.
+
+````math
+\mathcal P_{q\gets p}(X) = X\cdot\frac{q}{p}.
+````
+"""
+vector_transport_to(::PositiveNumbers, ::Any, ::Any, ::Any, ::ParallelTransport)
+function vector_transport_to(
+    ::PositiveNumbers,
+    p::Real,
+    X::Real,
+    q::Real,
+    ::ParallelTransport,
+)
+    return X * q / p
+end
+
+function vector_transport_direction(
+    M::PositiveNumbers,
+    p::Number,
```

Comment:
Why do you use sometimes `Number` and sometimes `Real`? I think using `Real` everywhere should be fine.

Reply:
You'Re right, I will unify that